### PR TITLE
[HLRC] Added support for CCR Stats API

### PR DESCRIPTION
**Pull request by martijnvg**
This change also adds documentation for the CCR Stats API.

This is PR is big, but that is mainly because indices follow and auto follow stats response classes and associated tests. These classes just have a lot of fields.

Relates to #33824

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ccr/IndicesFollowStats.java [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+
+public final class IndicesFollowStats {
+
+    static final ParseField INDICES_FIELD = new ParseField("indices");
+    static final ParseField INDEX_FIELD = new ParseField("index");
+    static final ParseField SHARDS_FIELD = new ParseField("shards");
+
+    private static final ConstructingObjectParser<Tuple<String, List<ShardFollowStats>>, Void> ENTRY_PARSER =
+        new ConstructingObjectParser<>("entry", args -> {
+        String index = (String) args[0];
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/ccr/IndicesFollowStats.java]: spacing is odd here, maybe tab in the closure a bit more

File: client/rest-high-level/src/test/java/org/elasticsearch/client/CCRIT.java [hunk 2]
```diff
@@ -148,8 +170,11 @@ public void testAutoFollowing() throws Exception {
         assertThat(response.isAcknowledged(), is(true));
 
         assertBusy(() -> {
-            assertThat(indexExists("copy-logs-20200101"), is(true));
+            CcrStatsRequest ccrStatsRequest = new CcrStatsRequest();
+            CcrStatsResponse ccrStatsResponse = execute(ccrStatsRequest, ccrClient::getCcrStats, ccrClient::getCcrStatsAsync);
+            assertThat(ccrStatsResponse.getAutoFollowStats().getNumberOfSuccessfulFollowIndices(), equalTo(1L));
         });
+        assertThat(indexExists("copy-logs-20200101"), is(true));
```
Comment (hub-cap) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/CCRIT.java]: how come you pulled this out of the assertBusy? just making sure it wasnt an omission.

**Review by hub-cap — APPROVED** [review]
small nits, id like to know the answer to a Q but i wont block.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/CCRIT.java (hunk 2 above)
Comment (martijnvg) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/CCRIT.java]: If the `numberOfSuccessfulFollowIndices` is 1 then it means that the follower index should already have been created. So there is no need to place the check whether `copy-logs-20200101` exists in the assertBusy. In fact if it isn't there immediately after `numberOfSuccessfulFollowIndices` reports 1 then it is a bug.

**Pull request merged**